### ISSUE TITLE
[Ticket Updated for 1.0B] EZ Flash Omega Definitive Edition compatibility issues

### ISSUE THREAD

**Issue by coolbho3k** [frederic-mahe/Analogue-Pocket#1]
1.0B EDIT:

1. With 1.0B, the EZ Flash Omega DE seems to be working better for many people. I can vouch for 1.0B having fixed the problem below for me.
2. One person reported a "Bad Header" BIOS error on 1.0B after saving with one game: https://twitter.com/noj86/status/1488623683740524545. Reverting to 1.0A fixed the issue for them.
3. For me, trying to boot from NOR in Mode B gives me a Link Mode error. Trying to boot from NOR from a menu still gives me a black screen.

Issues as of 1.0A:

A few possible issues with the [EZ Flash Omega Definitive Edition](https://www.ezflash.cn/product/ez-flash-omega-definitive-edition/) have been reported, in decreasing order of severity:

- Cart won't boot at all. Analogue BIOS Link-Mode. [Report 1](https://gbatemp.net/threads/ez-flash-omega-analogue-pocket-firmware.604607/post-9692400) | [Report 2](https://www.reddit.com/r/AnaloguePocket/comments/rqzhhi/ez_flash_omega_de_not_working_for_me_has_anyone/) | [Report 3](https://www.reddit.com/r/AnaloguePocket/comments/rqzhhi/ez_flash_omega_de_not_working_for_me_has_anyone/hqdhsuk/?utm_source=reddit&utm_medium=web2x&context=3) | [Report 4](https://www.reddit.com/r/AnalogueInc/comments/rkq0kw/ez_flash_omega_de_compatibility/?ref=share&ref_source=embed&utm_content=body&utm_medium=post_embed&utm_name=743b978e57f7497bb5cbb583e752f9af&utm_source=embedly&utm_term=rkq0kw)
- [Cart boots games, but crashes frequently.](https://www.reddit.com/r/AnalogueInc/comments/rorm08/compatibility_on_the_ez_flash_omega_de_and_the/)
- When working, **the NOR flash functionality on the cart seems to broken**. The cart boots in Mode A and games work well, but you get a white screen when the physical switch on the cart is in Mode B (which sets the cart to boot directly from NOR flash). You also cannot boot to a game in the NOR flash from the second tab in the menu.
- For me, everything else works except Mode B, and the EZ Flash in-game menu functionality for in-game reset and save states also freezes from time to time. It doesn't freeze on a real GBA.

EDIT 01/18/2022: **My WORKING cart stopped working and reports Analogue BIOS Link-Mode now**. This happened out of the blue - it was working the night before and suddenly stopped working, and I hadn't even swapped carts. I am able to get it to work by booting a real game, sleeping the Pocket, swapping for the flashcart, and waking the Pocket. It still works on a real GBA.

**Comment by coolbho3k** [comment]
It's worth noting that the EZ Flash team has [requested a Pocket unit to test with](https://gbatemp.net/threads/ez-flash-omega-analogue-pocket-firmware.604607/#post-9695796).

It appears they're in fulfillment Group C. I think it's worth it to send them a development unit as they are one (of the two) most popular current flashcart manufacturers.

**Comment by coolbho3k** [comment]
One person on GBAtemp got the Omega DE to work [by inserting a real cart, sleeping the Pocket, swapping to his flashcart, and unsleeping the Pocket](https://gbatemp.net/threads/ez-flash-omega-analogue-pocket-firmware.604607/post-9705901). This trick has been independently confirmed by one other person on the Classic Gaming Discord. 

Can anyone else who is experiencing the "Analogue BIOS Link Mode" screen with the Omega DE confirm if this fixes it?

**Comment by Malec1337** [comment]
I followed each step from the post you linked to and it worked for me. I was able to use my EZ Flash Omega DE this way.

**Comment by coolbho3k** [comment]
> I followed each step from the post you linked to and it worked for me. I was able to use my EZ Flash Omega DE this way.

That's awesome. This means it's fixable in software.

By the way, can you save state on the Omega DE's menu and then successfully load it when you get the white screen again?

**Comment by Malec1337** [comment]
Save State did not work for me. I started Bomberman 2 Blue on my DE with the steps mentioned earlier and went into the DE Save State menu (Start + L + R). The menu opened and i choose "save". After this the game was broken. The hole screen turned green. I was able to go into the Save State menu again and choose "load". After that my Pocket crashed. I had to reboot it.

**Comment by coolbho3k** [comment]
Omega DE save states and in general in-game menu is a hit or miss for me and definitely is more finicky than real hardware.

I've gotten it to work well in Pokemon Emerald but it didn't work so well in Mother 3.

**Comment by coolbho3k** [comment]
My Omega DE very suddenly stopped working with the Pocket today. It was working yesterday and up until now had been working since launch. I had left the cart in the Pocket and turned it on this afternoon and it wasn't working. I cleaned the contacts very thoroughly with contact cleaner and it still doesn't work.

Everything else seems to be fine. The Pocket will boot real games. And a real GBA will still boot the Omega DE.

The "sleep with a real game" workaround works to boot the Omega DE on the Pocket, but then it prompts me to update the kernel.

**Comment by Malec1337** [comment]
Strange. So it is possible that it will not work in the future even if it is working now. I hope EZ Flash and Analogue are able to fix this soon. My EZ Flash Junior works great with the Pocket (Newest Beta Firmware) and i hope my Omega DE will soon. I like my EZ Flashes :)

**Comment by coolbho3k** [comment]
1.0B has fixed the issue for me again. The Pocket lingers on the Analogue BIOS screen for a second unlike with other carts, leading me to believe that the firmware is now detecting the Omega DE and giving it a bit more time to initialize before booting. Anyone else want to try?

Boot from NOR doesn't work. Haven't tried Mode B yet.

**Comment by Malec1337** [comment]
I tried it and it works for me now. As you mentioned it stays a little bit longer in the BIOS screen and then boots the Omega DE.

**Comment by frederic-mahe** [comment]
> Boot from NOR doesn't work. Haven't tried Mode B yet.

hi @coolbho3k if you think the issue is solved, please close it. If the problem is partially solved, maybe it is best to open a new issue with a more specific description.

**Comment by noj86** [comment]
I confirm that 1.0B brought issue with my Omega DE, I reverted back to 1.0A to make it work again but with the latest firmware I get the Bad Header error systematically. I have found at last one other person getting the exact same issue as me since the 1.0B update. https://twitter.com/JoeSpaghet/status/1488622060200992779

I've reported more specifically the issue in this thread and tested against a regular non DE Omega and the Ez-Flah Junior too.
https://gbatemp.net/threads/analogue-pocket-fw-1-0b-ez-flash-omega-de-new-issue-after-first-boot.607082/
